include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,308 @@
+
+        while (is_char<'@'>(*ranges::begin(stream_view)))
+            detail::consume(stream_view | view::take_line);
+
+        /* ID */
+        std::string first; // Checking if id is given or set to *
+        ranges::copy(stream_view | view::take_until_or_throw(is_tab || is_blank), std::back_inserter(first));
+        if (first == "*")
+            throw parse_error{"The ID field may not be empty for SAM files."};
+
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            ranges::copy(first | view::char_to<value_type_t<id_type>>, std::back_inserter(id)); //Add first to id
+
+            if (options.truncate_ids)
+            {
+              detail::consume(stream_view | view::take_until_or_throw(is_tab));
+            }
+            else
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                       | view::char_to<value_type_t<id_type>>,
+                           std::back_inserter(id));
+            }
+        }
+        else
+        {
+             detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+
+        //Jump over SAM colums 2-9 (FLAG RNAME POS MAPQ CIGAR RNEXT PNEXT TLEN)
+        for (int i = 0; i < 8;i++)
+        {
+            ++begin(stream_view);
+            detail::consume(stream_view | view::take_until_or_throw(is_tab));
+        }
+        ++begin(stream_view);
+
+        auto seq_view = stream_view | view::take_until_or_throw(is_tab); // until next tab
+
+         // Sequence
+         if constexpr (!detail::decays_to_ignore_v<seq_type>)
+         {
+             auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
+             ranges::copy(seq_view | view::transform([is_legal_alph] (char const c) // enforce legal alphabet
+                                      {
+                                          if (!is_legal_alph(c))
+                                          {
+                                              throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                is_legal_alph.msg.string() +
+                                                                " evaluated to false on " +
+                                                                detail::make_printable(c)};
+                                          }
+                                          return c;
+                                      })
+                                    | view::char_to<value_type_t<seq_type>>, // convert to actual target alphabet
+                           std::back_inserter(sequence));
+
+              sequence_size_after = ranges::size(sequence);
+          }
+          else
+          {
+              for (auto it = begin(seq_view); it != end(seq_view); ++it)
+                  ++sequence_size_after;
+          }
+          ++begin(stream_view); // skip tab
+
+          std::string qual;
```

Comment:
Again do not use a temporary variable. Just query for `*` like for id and seq

Reply:
Another question about qualities: What if we have the quality length is greater than the sequence length? view::take_exactly_or_throw does accept this.